examples/demo-jobs/.neuro/demo.yml

```diff
@@ -29,55 +26,44 @@ tasks:
     bash:
       echo "This batch generates two numeric datasets and calculates the sum of averages."
 
-  - id: first_dataset
-    needs: [entry]
-    image: ${{ images.numpy.ref }}
-    volumes:
-      - ${{ volumes.dataset_1.ref_rw }}
-    python: |
-      import numpy
-      numpy.random.seed(${{ args.random_seed }})
-      dataset = numpy.random.rand(${{ args.size_first }})
-      numpy.save("/data/dataset.npy", dataset)
-
-  - id: second_dataset
+  - id: generate_dataset_${{ matrix.size }}
     needs: [entry]
     image: ${{ images.numpy.ref }}
+    strategy:
+      matrix:
+        # Problem #1: I can't list vars inside matrix.
```

Comment:
What variables are you talking about?

Reply:
It is about `args.size_first` and `args.size_second`.
The following is not valid yaml:
```
size: [${{ args.size_first }}, ${{ args.size_second }}]
```
It is because inside inline lists (json-style `[]`) the `{` has special meaning - it defines mapping. So our template sequence breaks yaml parser.

Reply:
That's perfectly fine; Github Workflows has the same limitation.
Use `size: ["${{ args.size_first }}", "${{ args.size_second }}"]` as an "official" workaround.